### PR TITLE
[GEAR-319] UTH Request

### PR DESCRIPTION
**Pull request by joshicola**
Incorporating UTH requests into recon-all.

### CODE REVIEW

**Comment by joshicola** [comment]
Subj is retrieved from hierarchy, see cancelled job:
https://ss.ce.flywheel.io/#/jobslog/job/5e728125529e1613b8812db5

File: bin/parse_config.py [hunk 1]
```diff
@@ -29,73 +43,138 @@ def parse_config(args):
...
+            max_cpus = mp.cpu_count()
+            # use of -parallel defaults to cpu_count unless specified
+            if context.config.get('n_cpus'):
+                # However, we will do strict checking on max_cpus
+                if context.config['n_cpus'] <= max_cpus:
+                    option_string += '-openmp ' + str(context.config['n_cpus'])
+                else:
+                    option_string += '-openmp ' + str(max_cpus)
+            else:
+                option_string += '-openmp ' + str(max_cpus)
```
Comment (lmperry) [inline bin/parse_config.py:81]: NIce.

File: bin/parse_config.py [hunk 2]
```diff
@@ -29,73 +43,138 @@ def parse_config(args):
...
+
+        # grab input, config, and project in order they are checked
+        # from input:
+        fs_license_file = context.get_input_path('freesurfer_license_file')
+        # from config:
+        fs_license = context.config.get('freesurfer_license')
+        # from project (needs api-key in config.json/manifest.json):
+        project_id = fw.get_analysis(destination_id).parents.project
+        project = fw.get_project(project_id)
+        project_license = project.get_file('license.txt')
```
Comment (lmperry) [inline bin/parse_config.py:132]: This is a completely sensible option, however it's not one of the [three "accepted" ways of providing a license](https://docs.flywheel.io/hc/en-us/articles/360013235453-How-to-include-a-Freesurfer-license-file-in-order-to-run-a-Freesurfer-or-fMRIPrep-gear-) (though it really should be). 
The third option is to provide the license as project context, or `info` on the project, and grab it from there: ```project.info.get('FREESURFER_LICENSE')```. Full example [here](https://github.com/flywheel-apps/fmriprep/blob/5.6.1_1.1.4/fs_license.py#L30-L36).

**Review by lmperry — CHANGES_REQUESTED** [review]
One change requested around license parsing from project context (`info`).

File: bin/parse_config.py (hunk 2 above)
Comment (joshicola) [reply bin/parse_config.py:132]: I will make it an option.

**Review by lmperry — APPROVED** [review]

**Pull request merged**